docs/rst/use_cases/keys/keys.rst

```diff
@@ -0,0 +1,183 @@
+
+Users can control the turtles using the keyboard, while the Qt application visually represents their movements.
+The Qt application terminal displays the turtles' positions, and the controller terminal shows the commands sent to each Turtle Node.
+The controller receives the turtles position.
+Each velocity and pose message is tagged with a key representing the turtle ID.
+
+The diagram below shows the entities required using topic keys:
+
+.. figure:: /rst/figures/use_cases/keys/keys.png
+    :align: center
+
+The diagram below shows the entities required **without** using topic keys:
+
+.. figure:: /rst/figures/use_cases/keys/non-keys.png
+    :align: center
+
+.. note::
+
+    Topic keys work exclusively with publishers and subscribers, not with services.
+    Multiple service clients can use the same service, but there can only be one service server for a given service.
+    Since each Turtle Node must have a service server to control goals, it does not make sense to have multiple clients (e.g., controllers) using the same service server with different keys.
+
+
+Prerequisites
+-------------
+
+Familiarity with the following topics is recommended:
+
+* :ref:`topic_keys`
+* `Content Filter Topic <https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/topic/contentFilteredTopic/contentFilteredTopic.html>`__
+
+It is required to have previously installed Vulcanexus using one of the following installation methods:
+
+* :ref:`linux_binary_installation`
+* :ref:`linux_source_installation`
+* :ref:`docker_installation`
+
+
+Keyed Messages
+--------------
+
+The keyed Topics use the following message types to represent the turtles' velocity and pose:
+
+``docs_turtlesim::msg::KeyedPose`` Represents the pose of a turtle with a key.
+
+- `key:` Identifier of the turtle.
+- `x:` X-coordinate of the turtle's position.
+- `y:` Y-coordinate of the turtle's position.
+- `theta:` Orientation angle of the turtle.
+- `linear_velocity:` Linear velocity of the turtle.
+- `angular_velocity:` Angular velocity of the turtle.
+
+.. code-block:: bash
+
+    // KeyedPose.idl
+    module docs_turtlesim {
+        module msg {
+            struct KeyedPose {
+                @key long key;
+                double x;
+                double y;
+                double theta;
+
+                double linear_velocity;
+                double angular_velocity;
+            };
+        };
+    };
+
+``docs_turtlesim::msg::KeyedVelocity`` Represents the velocity of a turtle with a key.
+
+- `key:` Identifier of the turtle.
+- `linear:` Linear velocity of the turtle.
+- `angular:` Angular velocity of the turtle.
+
+.. code-block:: bash
+
+    // KeyedTwist.idl
+    module docs_turtlesim {
+        module msg {
+            struct KeyedTwist {
+                @key long key;
+                docs_turtlesim::msg::Vector3  linear;
+                docs_turtlesim::msg::Vector3  angular;
+            };
+        };
+    };
```

Comment:
Same comments as before here. Remove the list of members and add the description of each member in a comment in the IDL